testing-common/src/main/groovy/io/opentelemetry/auto/test/AgentTestTrait.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.auto.test
+
+import com.google.common.base.Predicate
+import groovy.transform.stc.ClosureParams
+import groovy.transform.stc.SimpleType
+import io.opentelemetry.auto.test.asserts.InMemoryExporterAssert
+import io.opentelemetry.sdk.trace.data.SpanData
+
+/**
+ * A trait which initializes agent tests, including bytecode manipulation and a test span exporter.
+ * All agent tests should implement this trait.
+ */
+trait AgentTestTrait {
```

Comment:
Somewhere should be documentation about `childSetupSpec` and `childSetup` methods